### ISSUE TITLE
System error: read parent: connection reset by peer

### ISSUE THREAD

**Issue by reynoldsalec** [kalabox/kalabox#456]
On running "kbox create pantheon" I was receiving the following error:

```
Error while installing app: "cvt": HTTP code is 500 which indicates error: server error - Cannot start container 81d10ec1c6c99f7f3a12f8afe3b553144ad48aa027deea66d78690721d90ada9: [8] System error: read parent: connection reset by peer

Error: HTTP code is 500 which indicates error: server error - Cannot start container 81d10ec1c6c99f7f3a12f8afe3b553144ad48aa027deea66d78690721d90ada9: [8] System error: read parent: connection reset by peer

    at /Users/alec/workspace/kalabox/node_modules/kalabox-engine-docker/node_modules/dockerode/node_modules/docker-modem/lib/modem.js:204:17
    at getCause (/Users/alec/workspace/kalabox/node_modules/kalabox-engine-docker/node_modules/dockerode/node_modules/docker-modem/lib/modem.js:232:7)
    at Modem.buildPayload (/Users/alec/workspace/kalabox/node_modules/kalabox-engine-docker/node_modules/dockerode/node_modules/docker-modem/lib/modem.js:203:5)
    at IncomingMessage.<anonymous> (/Users/alec/workspace/kalabox/node_modules/kalabox-engine-docker/node_modules/dockerode/node_modules/docker-modem/lib/modem.js:179:14)
    at IncomingMessage.EventEmitter.emit (events.js:117:20)
    at _stream_readable.js:919:16
    at process._tickCallback (node.js:419:13)
```

Error occurred upon attempting to pull down application code from Pantheon.

Tried restarting kalabox (after destroying the app), no dice.

Upon trying a different name for the app, the command worked.

**Comment by pirog** [comment]
So i am *pretty sure* this happens when you have an old entry in the appRegistry that does not correspond to an actual app that exists. Just need to add in some checks.

**Comment by reynoldsalec** [comment]
@pirog tried deleting the old (already destroyed) apps from the appRegistry.json file and that didn't resolve my issue, any ideas on further locations that the old app names might be getting stored?

**Comment by pirog** [comment]
@bcauldwell in addition to this we probably want to also 

1. Remove the entry from the appRegistry on kbox destroy
2. Prompt the user on destroy that WE ARE GOING TO DESTROY EVERYTHING and make them confirm that this is ok.
3. Also destroy the app folder.

**Comment by reynoldsalec** [comment]
It looks like the container is being removed before commands can be run on it, hence the 500:

```
debug: DOCKER =>  Created RUN container.
debug: DOCKER =>  Attaching to RUN container. stream=true, stdin=true, stdout=true, stderr=true
debug: DOCKER =>  Attached to RUN container.
debug: DOCKER =>  Starting RUN container. Binds=[/Users/alec:/ssh], VolumesFrom=[]
debug: DOCKER =>  Removing RUN container. force=true
debug: DOCKER =>  Removed RUN container.
```

Error occurs after.

**Comment by pirog** [comment]
This is what the normal output looks like

Container starting up
```
debug: DOCKER =>  Created RUN container.
debug: DOCKER =>  Attaching to RUN container. stream=true, stdin=true, stdout=true, stderr=true
debug: DOCKER =>  Attached to RUN container.
debug: DOCKER =>  Starting RUN container. Binds=[/Users/mpirog:/ssh], VolumesFrom=[]
debug: DOCKER =>  Started RUN container.
debug: DOCKER =>  Waiting on RUN container.
```

Running the actual command in the container
```
Warning: Permanently added '[codeserver.dev.1e821c4a-ac22-4187-9718-228ec1d5f190.drush.in]:2222,[166.78.107.76]:2222' (RSA) to the list of known hosts.
From ssh://codeserver.dev.1e821c4a-ac22-4187-9718-228ec1d5f190.drush.in:2222/~/repository
 * branch            master     -> FETCH_HEAD
Already up-to-date.
```

Destroying the container 
```
debug: DOCKER =>  Removing RUN container. force=true
debug: DOCKER =>  Removed RUN container.
```

**Comment by pirog** [comment]
@bcauldwell i think this issue is very @reynoldsalec's specific so im going to icebox until we hear about it from someone else.

**Comment by alphex** [comment]
This isn't unique to @reynoldsalec 
I'm having issues with it right now.

With my previous ticket #530 related to #536 I then (foolishly), simply tried to just RM -RF the application I built.

I then tried to re-create it, and got, as I should have thought, logical conflcts.  The configuration files think the directory still has things in it, because I didn't "kbox destroy" it.

I created a new application name (new directory was created).
(and this is a different problem from this thread), it won't work either.

So I tried to do it the right way.

CD into the app.
I can see the kalabox.json and dockerfiles/ ... and I type "kbox destroy"
my EXPECTATION is that this would wipe out this directory and any history of it on my system.

This is what I get.
https://gist.github.com/alphex/d84f870ac786aa39a125

1. CD in to app
2. "kbox destroy"
it says it's "removing the codez"
returns me to prompt.
I then "ls -la" and I still see all of the "codez" I expected to be removed.

**Comment by pirog** [comment]
@alphex nice! well that was quick. Can you confirm that you are actually getting the same 500 error that @reynoldsalec posted?

So if you get a fail during the site-pull-down part of kbox create as in the case of the non-existent database (#530) you can actually go into the app directory after it fails and run a `kbox pull` to try again. 

So as it relates to `kbox destroy` what is actually happening here is it will remove all of the containers running in the VM that support your app. It will also remove any syncthing folders that are set up. It will also remove the "code" directory which is where your sites git code would live. (this is the removing the codez part). You don't have this dir in your gist because you presumably never started the app. But it would be at `/code` inside of your app. 

As this issue indicates we think also removing ALL of the code for the ENTIRE app would be cleaner so i def +1 that. 

All that said, because you deleted the entire app folder before doing a kbox destroy you are probably getting errors because kbox is still looking for stuff that doesn't exist. It's probably likely other people will try to do this as well so we should try to handle this scenario. 

In the meantime i would go into `~/.kalabox/` and look for a file called appRegistry.json and manually remove whatever the offending app is and try the whole thing again. Note that you now have orphaned containers for a previous app so if you name it something similar it will probably fail at some point. We should try to handle this scenario as well.

If you could help report back on all of the above i can create separate tickets as needed.

**Comment by pirog** [comment]
Going to kick this to alpha4 until we can get more feedback on how to replicate.

**Comment by pirog** [comment]
@reynoldsalec are you still getting this at all?

**Comment by bcauldwell** [comment]
I can't fix this with the methods people have been using at https://github.com/docker/docker/issues/14203. I'm going to move to another issue and come back to this one.

**Comment by pirog** [comment]
k, will move into next sprint @bcauldwell

**Comment by pirog** [comment]
Let's do research on whether 1.9.1 fixes this.

to replicate: 
pull down playbox but name it playbox2

**Comment by pirog** [comment]
@bcauldwell just pulled down playbox and named it playbox2 but was unable to replicate. anything else i need to do here?

**Comment by pirog** [comment]
@bcauldwell here is the updated iso image. To update manually you will want to 

```
kbox down
cd ~/.kalabox/.provider
mv boot2docker.iso boot2docker.iso.old

kbox up
boot2docker --vm="Kalabox2" ssh

Boot2Docker version 1.9.1, build master : cef800b - Fri Nov 20 20:15:58 UTC 2015
Docker version 1.9.1, build a34a1d5

```

**Comment by pirog** [comment]
@bcauldwell im going to say we need to do the following here before we close this out

- [ ] Provide an upgrade pathway to grab the new boot2docker image
